src/script/run-make.sh

```diff
@@ -0,0 +1,159 @@
+            echo "Please install bc and re-run." 
+            exit 1
+        fi
+        if test "$(echo "$VERSION_ID >= 22" | bc)" -ne 0; then
+            install_cmd="dnf -y install"
+        else
+            install_cmd="yum install -y"
+        fi
+    elif type zypper > /dev/null 2>&1 ; then
+        install_cmd="zypper --gpg-auto-import-keys --non-interactive install --no-recommends"
+    elif type apt-get > /dev/null 2>&1 ; then
+        install_cmd="apt-get install -y"
+        which_pkg="debianutils"
+    fi
+
+    if ! type sudo > /dev/null 2>&1 ; then
+        echo "Please install sudo and re-run. This script assumes it is running"
+        echo "as a normal user with the ability to run commands as root via sudo." 
+        exit 1
+    fi
+    if [ -n "$install_cmd" ]; then
+        $DRY_RUN sudo $install_cmd ccache $which_pkg
+    else
+        echo "WARNING: Don't know how to install packages" >&2
+        echo "This probably means distribution $ID is not supported by run-make-check.sh" >&2
+    fi
+
+    if ! type ccache > /dev/null 2>&1 ; then
+        echo "ERROR: ccache could not be installed"
+        exit 1
+    fi
+
+    if test -f ./install-deps.sh ; then
+	    export WITH_SEASTAR=1
+	    $DRY_RUN source ./install-deps.sh || return 1
+        trap clean_up_after_myself EXIT
+    fi
+
+    cat <<EOM
+Note that the binaries produced by this script do not contain correct time
+and git version information, which may make them unsuitable for debugging
+and production use.
+EOM
+    save_ccache_conf
+    # remove the entropy generated by the date/time embedded in the build
+    $DRY_RUN export SOURCE_DATE_EPOCH="946684800"
+    $DRY_RUN ccache -o sloppiness=time_macros
+    $DRY_RUN ccache -o run_second_cpp=true
+    if [ -n "$JENKINS_HOME" ]; then
+        # Build host has plenty of space available, let's use it to keep
+        # various versions of the built objects. This could increase the cache hit
+        # if the same or similar PRs are running several times
+        $DRY_RUN ccache -o max_size=100G
+    else
+        echo "Current ccache max_size setting:"
+        ccache -p | grep max_size
+    fi
+    $DRY_RUN ccache -sz # Reset the ccache statistics and show the current configuration
+}
+
+function configure() {
+    CMAKE_BUILD_OPTS="$@"
+    CMAKE_BUILD_OPTS+=$(detect_ceph_dev_pkgs)
+    $DRY_RUN ./do_cmake.sh $CMAKE_BUILD_OPTS $@ || return 1
+}
+
+function build() {
+    local targets="$@"
+    $DRY_RUN cd build
+    BUILD_MAKEOPTS=${BUILD_MAKEOPTS:-$DEFAULT_MAKEOPTS}
+    test "$BUILD_MAKEOPTS" && echo "make will run with option(s) $BUILD_MAKEOPTS"
+    $DRY_RUN make $BUILD_MAKEOPTS $targets || return 1
+    $DRY_RUN ccache -s # print the ccache statistics to evaluate the efficiency
+}
+
+DEFAULT_MAKEOPTS=${DEFAULT_MAKEOPTS:--j$(get_processors)}
+
+if [ "$0" = "$BASH_SOURCE" ]; then
+    # not sourced
+    if [ `uname` = FreeBSD ]; then
+        GETOPT=/usr/local/bin/getopt
```

Comment:
OK, we're doing the same in other scripts as well (e.g. `src/tools/setup-virtualenv.sh`).